resources/examples/manager/BasicAssetLibrary/schema.json

```diff
@@ -0,0 +1,41 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
```

Comment:
[question] Neat that we're adding a JSON schema, but curious what prompted this? Do we expect BAL to be used externally?  Or is it just that it seems good practice for completeness?

Reply:
We will use it in `otio-openassetio` for e2e tests, and any host implementation may wish to use it for simple, repeatable test-harness workflows. As such, it felt like it would be better than writing loose documentation.